docs/pages/docs/handbook/sharing-code/internal-packages.mdx

```diff
@@ -0,0 +1,255 @@
+
+## What makes a package _internal?_
+
+External packages **run their files through a bundler** before putting them on a package registry. This means they need a _lot_ of tooling to handle.
+
+- **Bundlers**: to build the package
+- **Versioning**: to help with versioning and releases
+- **Publishing**: to publish the package
+
+If you want to use those files locally, you'll also need:
+
+- **Dev scripts**: for bundling the package locally when files change
+
+Because internal packages are not published, we can skip _all_ of these steps. Instead of bundling our package ourselves, we're going to make the **app which imports the package bundle it for us**.
+
+This sounds complex, but it's extremely easy to set up.
+
+## Our first internal package
+
+We're going to create a shared `math-helpers` package inside our monorepo.
+
+### 1. Create your monorepo
+
+If you don't have an existing monorepo, create one [using our guide](/docs/getting-started/create-new).
+
+### 2. Create a new package
+
+Inside `/packages`, create a new folder called `math-helpers`.
+
+```bash
+mkdir packages/math-helpers
+```
+
+Create a `package.json`:
+
+```json filename="packages/math-helpers/package.json"
+{
+  "name": "math-helpers",
+  "dependencies": {
+    // Use whatever version of TypeScript you're using
+    "typescript": "latest"
+  }
+}
+```
+
+Create a `src` folder, and add a TypeScript file at `packages/math-helpers/src/index.ts`.
+
+```ts filename="packages/math-helpers/src/index.ts"
+export const add = (a: number, b: number) => {
+  return a + b;
+};
+
+export const subtract = (a: number, b: number) => {
+  return a - b;
+};
+```
+
+You'll also need to add a `tsconfig.json` at `packages/math-helpers/tsconfig.json`:
+
+```json filename="packages/math-helpers/tsconfig.json"
+{
+  "compilerOptions": {
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
+    "moduleResolution": "node",
+    "preserveWatchOutput": true,
+    "skipLibCheck": true,
+    "noEmit": true,
+    "strict": true
+  },
+  "exclude": ["node_modules"]
+}
+```
+
+Great! We've now got all the files we need for our internal package.
+
+### 3. Import the package
+
+We're now going to import the package and see what happens. Go into one of your apps, and add `math-helpers` into the dependencies of its package.json:
+
+<Tabs items={['npm', 'yarn', 'pnpm']} storageKey="selected-pkg-manager">
+  <Tab>
+```jsonc filename="apps/web/package.json"
+{
+  "dependencies": {
+    "math-helpers": "*"
+  }
+}
+```
+  </Tab>
+  <Tab>
+```jsonc filename="apps/web/package.json"
+{
+  "dependencies": {
+    "math-helpers": "*"
+  }
+}
+```
+  </Tab>
+  <Tab>
+```jsonc filename="apps/web/package.json"
+{
+  "dependencies": {
+    "math-helpers": "workspace:*"
+  }
+}
+```
+  </Tab>
+</Tabs>
+
+[Install all packages from root](/docs/handbook/package-installation) to ensure that dependency works.
```

Comment:
For my learning too... is editing `package.json` better than running `npm i <package>` from inside the workspace? Or something like `npm i -w <workspace> <package>` from root? It may be better to document CLI commands if rather than editing files, if so, as there's less room for error!

Reply:
I think any of these is fine!